internal/webhooks/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -109,8 +109,26 @@ func (webhook *BareMetalHost) validateChanges(oldObj *metal3api.BareMetalHost, n
 		errs = append(errs, errors.New("BMC address can not be changed if the BMH is not in the Registering state, or if the BMH is not detached"))
 	}
 
-	if oldObj.Spec.BootMACAddress != "" && newObj.Spec.BootMACAddress != oldObj.Spec.BootMACAddress {
-		errs = append(errs, errors.New("bootMACAddress can not be changed once it is set"))
+	if oldObj.Spec.BootMACAddress != "" {
+		// BootMACAddress is immutable once set - cannot be changed or cleared
+		if newObj.Spec.BootMACAddress == "" {
+			errs = append(errs, errors.New("bootMACAddress can not be unset once it is set"))
+		} else {
+			// Compare MAC addresses in a case-insensitive manner by parsing them.
+			oldMAC, oldErr := net.ParseMAC(oldObj.Spec.BootMACAddress)
+			newMAC, newErr := net.ParseMAC(newObj.Spec.BootMACAddress)
+
+			// If the old MAC is invalid, we cannot reliably verify immutability.
+			// The new MAC is validated by validateBMCAccess (called via validateHost above).
+			if oldErr != nil {
+				errs = append(errs, fmt.Errorf("cannot validate bootMACAddress immutability: existing value %q is not a valid MAC address", oldObj.Spec.BootMACAddress))
+			}
+
+			// If both MACs are valid and different, enforce immutability.
+			if oldErr == nil && newErr == nil && oldMAC.String() != newMAC.String() {
+				errs = append(errs, errors.New("bootMACAddress can not be changed once it is set"))
+			}
+		}
```

Comment:
```suggestion
    if oldObj.Spec.BootMACAddress != "" && newObj.Spec.BootMACAddress != oldObj.Spec.BootMACAddress {
		errs = append(errs, errors.New("bootMACAddress can not be changed once it is set"))
	} else {
	    newMAC, newErr := net.ParseMAC(newObj.Spec.BootMACAddress)
	    if newErr == nil && oldMAC.String() != newMAC.String() {
				errs = append(errs, errors.New("bootMACAddress can not be changed once it is set"))
		}
	}
```

Reply:
I think I much better understand your case now, and I think a lot of complexity is introduced here with using net.ParseMac.

Your primary objective is to make sure that it is accepted when the the old mac is changed to a new  mac what is essentially the same but with a different capitalization/case.

My objective is to have the simplest admission control and not do things like handle errors related to the old mac, what if we do this?:
  ```suggestion
  if oldObj.Spec.BootMACAddress != "" && strings.ToLower(newObj.Spec.BootMACAddress) != strings.ToLower(oldObj.Spec.BootMACAddress) {
		errs = append(errs, errors.New("bootMACAddress can not be changed once it is set"))
  }
```